static/js/composebox_typeahead.js

```diff
@@ -349,9 +349,13 @@ exports.content_typeahead_selected = function (item) {
                 + '#**' + item.name + '** ');
         $(document).trigger('streamname_completed.zulip', {stream: item});
     } else if (this.completing === 'syntax') {
-        rest = "\n" + beginning.substring(beginning.length - this.token.length - 4,
-                beginning.length - this.token.length).trim() + rest;
-        beginning = beginning.substring(0, beginning.length - this.token.length) + item + "\n";
+        if (rest === '') {
+            rest = "\n" + beginning.substring(beginning.length - this.token.length - 4,
+                    beginning.length - this.token.length).trim() + rest;
+            beginning = beginning.substring(0, beginning.length - this.token.length) + item + '\n';
```

Comment:
Can we try to make this code more readable and/or add comments?  If you don't know what token and item are supposed to look like here, it's very cryptic.